Problem: upgrade to ethers 2.0 fails

```
error[E0277]: `?` couldn't convert the error to `transaction::ethereum::error::EthError`
[347](https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/4347558539/jobs/7595014943#step:4:348)
  --> common/src/transaction/ethereum.rs:73:65
[348](https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/4347558539/jobs/7595014943#step:4:349)
   |
[349](https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/4347558539/jobs/7595014943#step:4:350)
73 |             EthNetwork::Known { name } => Chain::from_str(&name)?,
[350](https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/4347558539/jobs/7595014943#step:4:351)
   |                                                                 ^ the trait `From<strum::ParseError>` is not implemented for `transaction::ethereum::error::EthError`
[351](https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/4347558539/jobs/7595014943#step:4:352)
   |
[352](https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/4347558539/jobs/7595014943#step:4:353)
   = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
[353](https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/4347558539/jobs/7595014943#step:4:354)
   = help: the following other types implement trait `From<T>`:
[354](https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/4347558539/jobs/7595014943#step:4:355)
             <transaction::ethereum::error::EthError as From<TryFromPrimitiveError<ethers::ethers_addressbook::Chain>>>
[355](https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/4347558539/jobs/7595014943#step:4:356)
             <transaction::ethereum::error::EthError as From<ethers::abi::Error>>
[356](https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/4347558539/jobs/7595014943#step:4:357)
             <transaction::ethereum::error::EthError as From<ethers::ethers_contract::AbiError>>
[357](https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/4347558539/jobs/7595014943#step:4:358)
             <transaction::ethereum::error::EthError as From<transaction::ethereum::error::Eip712Error>>
[358](https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/4347558539/jobs/7595014943#step:4:359)
   = note: required for `std::result::Result<ethers::ethers_addressbook::Chain, transaction::ethereum::error::EthError>` to implement `FromResidual<std::result::Result<Infallible, strum::ParseError>>`
[359](https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/4347558539/jobs/7595014943#step:4:360)
```
https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/4347558539/jobs/7595014943

--- a/bindings/cpp/Cargo.toml
+++ b/bindings/cpp/Cargo.toml
@@ -14,7 +14,7 @@ anyhow = "1"
 serde="1"
 serde_json="1"
 siwe = { version = "0.5" }
-ethers = { version = "1", features = ["rustls"] }
+ethers = { version = "2", features = ["rustls"] }
 hex = "0.4"
 tokio = { version = "1", features = ["rt"] }
 

--- a/bindings/wasm/Cargo.toml
+++ b/bindings/wasm/Cargo.toml
@@ -35,7 +35,7 @@ tendermint = "0.29"
 
 [dev-dependencies]
 wasm-bindgen-test = "0.3"
-ethers = { version = "1", features = ["rustls"] }
+ethers = { version = "2", features = ["rustls"] }
 wasm-timer = "0.2"
 tendermint-rpc = "0.29"
 defi-wallet-core-proto = { version = "0.1", path = "../../proto" }

--- a/common/Cargo.toml
+++ b/common/Cargo.toml
@@ -31,7 +31,7 @@ bip39 = { version = "1", default-features = false }
 # FIXME: switch to upstream crates.io when released
 cosmrs = { git = "https://github.com/crypto-com/cosmos-rust.git" }
 eyre = "0.6"
-ethers = { version = "1", features = ["rustls", "abigen"] }
+ethers = { version = "2", features = ["rustls", "abigen"] }
 ibc = { version = "0.31", features = ["serde"], default-features = false }
 ibc-proto = { version = "0.26", default-features = false }
 itertools = "0.10"

--- a/common/src/common.udl
+++ b/common/src/common.udl
@@ -410,6 +410,7 @@ enum EthError {
   "ContractCallError",
   "SignatureError",
   "ChainidError",
+  "IncorrectChainidError",
   "AbiError",
   "DynamicAbiError",
   "Eip712Error",

--- a/common/src/transaction/ethereum.rs
+++ b/common/src/transaction/ethereum.rs
@@ -70,7 +70,8 @@ impl TryFrom<EthNetwork> for Chain {
             EthNetwork::BSC => Chain::BinanceSmartChain,
             EthNetwork::Cronos => Chain::Cronos,
             EthNetwork::Polygon => Chain::Polygon,
-            EthNetwork::Known { name } => Chain::from_str(&name)?,
+            EthNetwork::Known { name } => Chain::from_str(&name)
+                .map_err(|e| EthError::IncorrectChainidError(e.to_string()))?,
             EthNetwork::Custom { chain_id, .. } => Chain::try_from(chain_id)?,
         })
     }

--- a/common/src/transaction/ethereum/error.rs
+++ b/common/src/transaction/ethereum/error.rs
@@ -42,6 +42,8 @@ pub enum EthError {
     SignatureError,
     #[error("Chainid error: {0}")]
     ChainidError(#[from] ParseChainError),
+    #[error("Incorrect chain name error: {0}")]
+    IncorrectChainidError(String),
     #[error("ABI error: {0}")]
     AbiError(#[from] abi::Error),
     #[error("Dynamic ABI error: {0}")]

